lib/util/git/private-key.ts

```diff
@@ -114,8 +117,32 @@ class SSHKey extends PrivateKey {
   }
 }
 
+class X509Key extends PrivateKey {
+  protected readonly gpgFormat = 'x509';
+
+  constructor(key: string) {
+    super(key.trim());
+  }
+
+  protected async importKey(): Promise<string | undefined> {
+    const keyFileName = upath.join(os.tmpdir() + '/git-private-x509.key');
+    await fs.outputFile(keyFileName, this.key);
+    const { stdout, stderr } = await exec(['gpgsm', '--import', keyFileName]);
```

Comment:
is the `gpgsm` part of a git / gnupg installation?

it seems to be a separate package, maybe we need to add it to our images first?

https://packages.debian.org/de/sid/gpgsm

Reply:
It seems already in our image